### PR TITLE
[MDS] Add dashboard assistant tests with multiple data source enabled

### PR DESCRIPTION
**Pull request by yubonluo**
### Description

Add dashboard assistant tests with multiple data source enabled

### Issues Resolved

[List any issues this PR will resolve]

### Check List

- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by raintygao** [comment]
https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/1361 Is this still opened?

**Comment by yubonluo** [comment]
> #1361 Is this still opened?

@raintygao I will close #1361.

File: cypress/integration/plugins/dashboards-assistant/mds_conversation_history_spec.js [hunk 1]
```diff
@@ -9,8 +9,18 @@ if (Cypress.env('DASHBOARDS_ASSISTANT_ENABLED')) {
   describe('Assistant conversation history spec', () => {
     let restoreShowHome;
     let restoreNewThemeModal;
+    let dataSourceId;
 
     before(() => {
+      if (Cypress.env('DATASOURCE_MANAGEMENT_ENABLED')) {
```
Comment (wanglam) [inline cypress/integration/plugins/dashboards-assistant/mds_conversation_history_spec.js:15]: Can we reuse the `setDefaultDataSourceForAssistant`? Maybe we can return the created data source id in `setDefaultDataSourceForAssistant` result.

File: cypress/utils/plugins/dashboards-assistant/commands.js [hunk 2]
```diff
@@ -193,13 +196,34 @@ Cypress.Commands.add('stopDummyServer', () => {
   });
 });
 
-Cypress.Commands.add('sendAssistantMessage', (body) =>
-  apiRequest(`${BASE_PATH}${ASSISTANT_API.SEND_MESSAGE}`, 'POST', body)
-);
+Cypress.Commands.add('sendAssistantMessage', (body, dataSourceId) => {
+  const url = `${BASE_PATH}${ASSISTANT_API.SEND_MESSAGE}${
```
Comment (wanglam) [inline cypress/utils/plugins/dashboards-assistant/commands.js:200]: Shall we use `qs` argument in `apiRequest`?

File: cypress/utils/plugins/dashboards-assistant/commands.js [hunk 3]
```diff
@@ -193,13 +196,34 @@ Cypress.Commands.add('stopDummyServer', () => {
...
+});
 
-Cypress.Commands.add('deleteConversation', (conversationId) =>
-  apiRequest(
-    `${BASE_PATH}${ASSISTANT_API.CONVERSATION}/${conversationId}`,
-    'DELETE'
-  )
-);
+Cypress.Commands.add('deleteConversation', (conversationId, dataSourceId) => {
+  const url = `${BASE_PATH}${ASSISTANT_API.CONVERSATION}/${conversationId}${
```
Comment (wanglam) [inline cypress/utils/plugins/dashboards-assistant/commands.js:207]: The same as above, prefer using `qs` argument in `apiRequest`.

File: cypress/utils/plugins/dashboards-assistant/commands.js (hunk 2 above)
Comment (yubonluo) [reply cypress/utils/plugins/dashboards-assistant/commands.js]: Sure, the code had been updated.

File: cypress/utils/plugins/dashboards-assistant/commands.js (hunk 3 above)
Comment (yubonluo) [reply cypress/utils/plugins/dashboards-assistant/commands.js]: Sure, the code had been updated.

File: cypress/integration/plugins/dashboards-assistant/mds_conversation_history_spec.js (hunk 1 above)
Comment (yubonluo) [reply cypress/integration/plugins/dashboards-assistant/mds_conversation_history_spec.js]: Sure, the code had been updated.

**Review by wanglam — APPROVED** [review]
LGTM

File: cypress/utils/plugins/dashboards-assistant/commands.js [hunk 4]
```diff
@@ -135,7 +135,10 @@ Cypress.Commands.add('registerRootAgent', () => {
 });
 
 Cypress.Commands.add('putRootAgentId', (agentId) => {
-  if (Cypress.env('SECURITY_ENABLED')) {
+  if (
+    Cypress.env('SECURITY_ENABLED') &&
+    !Cypress.env('DATASOURCE_MANAGEMENT_ENABLED')
```
Comment (raintygao) [inline cypress/utils/plugins/dashboards-assistant/commands.js:140]: Can you elaborate more why we add this line?

Comment (yubonluo) [reply cypress/utils/plugins/dashboards-assistant/commands.js:141]: Sure, the annotation has been added.

Comment (raintygao) [reply cypress/utils/plugins/dashboards-assistant/commands.js:141]: Thanks, I see, that makes sense.

**Review by raintygao — APPROVED** [review]

**Pull request merged**